api/types/kustomization.go

```diff
@@ -332,3 +359,101 @@ func (k *Kustomization) Unmarshal(y []byte) error {
 	*k = nk
 	return nil
 }
+
+// splitPatchStrategicMerge splits a single PatchStrategicMerge file into multiple PatchStrategicMerge files, if the
+// file contains multiple documents separated by the yaml separator.
+func splitPatchStrategicMerge(p PatchStrategicMerge, fSys filesys.FileSystem) ([]PatchStrategicMerge, error) {
```

Comment:
nit: I think it might be easier to follow if we _didn't_ use the `PatchStrategicMerge` type here. In the Kustomization struct, that type can represent either a path or a complete patch, but by the time we reach this method, we know what we have is a path (and what we return is a list of paths). Same for the other one: we know what we have in that case is one or more raw patches.

Reply:
Agreed.  It is unnecessary at this level and in fact adds some complexity, let me see if there is a way to refactor this.